src/uncrustify.cpp

```diff
@@ -69,11 +69,18 @@
 
 // VS throws an error if an attribute newer than the requested standard level
 // is used; everyone else just ignores it (or warns) like they are supposed to
-#if __cplusplus >= 201703L || !defined (_MSC_VER)
+
+#if __cplusplus >= 201703L
+#define NODISCARD    [[nodiscard]]
+#elif defined (__has_cpp_attribute)
```

Comment:
Huh... I didn't know that was a macro.

Reply:
According to [cppreference.com - feature_test](https://en.cppreference.com/w/cpp/experimental/feature_test) it is.